### DISCUSSION TITLE
How can I make a label change when a slider value changes

### DISCUSSION THREAD

**Discussion by sryu1** [TomSchimansky/CustomTkinter#744] [General]
Can someone please provide an example code snippet of a label changing every time the slider value changes? I managed to get something working, but it doesn't change the label but adds more labels which I don't want. Thanks :).

**Comment by avalon60** [comment]
You need a callback function. In the label widget instantiation, include command=my_callback:

`self.label = customtkinter.CTkLabel(master=...., command=my_callback)`

Use the callback function to configure the label text:

```
def my_callback(self):
    slider_value = self.my_slider.get()
    self.label.configure(text=f'new  value: {slider_value}')
```
Typically, the callback function will include a slider.get() call to obtain the value as it changes.

**Reply by sryu1** [reply]
ah ok, thank you! Also, is it possible for a text to be inside a progressbar? (I doubt that function exists)

**Reply by avalon60** [reply]
Not the last time I looked (4.6). I usually use a label widget adjacent to the slider.

**Reply by sryu1** [reply]
Ah, ok, thank you.

**Comment by fuzzyfazzy** [comment]
Hello - my question is opposite to this in a way, how do you input a value manually and make the slider move accordingly?

**Reply by avalon60** [reply]
Hi @fuzzyfazzy. The slider has a set() method. So if you entered a value to a CTkEntry field, you would probably want to bind the enter key event on the entry field, so that it executes a callback function, which grabs the value, from the entry field. 

E.g `entry_value = entry_field.get()` and then does `my_slider.set(entry_value)`.

To bind to the entry widget, you would need something like:

`entry_field.bind("<Return>", get_value_update_slider)`

So get_value_update_slider(), would execute the get and set commands described.

**Reply by fuzzyfazzy** [reply]
Thankyou very much!
